### PR TITLE
Issue #7567: Adding code examples for default configuration of MethodNameCheck

### PR DESCRIPTION
**Pull request by Vyom-Yadav**
Issue #7567: Adding code examples for default configuration of MethodNameCheck

### CODE REVIEW

**Comment by Vyom-Yadav** [comment]
@romani Kindly Check.

**Comment by Vyom-Yadav** [comment]
@romani Kindly review.

**Comment by Vyom-Yadav** [comment]
@romani Why this error is coming? I am unable to find the reason.

**Comment by Abhishek-kumar09** [comment]
Write Issue # in your PR description.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java [hunk 1]
```diff
@@ -137,6 +137,25 @@
  * }
  * </pre>
  * <p>
+ * An example of the default configuration for all methods:
+ * </p>
+ * <pre>
+ * &lt;module name="MethodName"&gt;
+ *    &lt;property name="format" value="^[a-z](_?[a-zA-Z0-9]+)*$"/&gt;
+ * &lt;/module&gt;
+ * </pre>
```
Comment (Abhishek-kumar09) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java:146]: You had to put code example for default configuration.
The default configuration is one without any other property.
This is not the default configuration since it has `format` property being explicitly written.
This is the default config:
```
<module name="MethodName"/>
```
Take help from here: https://checkstyle.sourceforge.io/config_naming.html#MethodName
See what are the default values for each property and then make some code examples accordingly.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java [hunk 2]
```diff
@@ -64,7 +64,6 @@
  * <pre>
  * &lt;module name="ConstantName"/&gt;
  * </pre>
- *
```
Comment (Abhishek-kumar09) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java:67]: Please revert this change.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java [hunk 3]
```diff
@@ -137,6 +137,25 @@
...
+ * class MyClass {
+ *   public void first_Method1() {} // OK
+ *   protected void second_Method2() {} // OK
+ *   private void Third_Method3() {} // violation, name 'ThirdMethod' must match
+ *                                 // pattern '^[a-z](_?[a-zA-Z0-9]+)*$'
+ *   void Fourth_Method4() {} // violation, name 'FourthMethod' must match
+ *                          // pattern '^[a-z](_?[a-zA-Z0-9]+)*$'
+ * }
+ * </pre>
+ * <p>
```
Comment (Abhishek-kumar09) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java:158]: If its an example of default config then write it below Line no. 85 in this file, since there you have to add code examples of default config.

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]
Some items.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java (hunk 2 above)
Comment (Vyom-Yadav) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java:67]: This change was I did in my master repo them I branched, so this is the original one only.

**Comment by Vyom-Yadav** [comment]
github generate site

**Comment by Vyom-Yadav** [comment]
github generate site

**Comment by Vyom-Yadav** [comment]
github generate site

**Comment by Vyom-Yadav** [comment]
github generate site

**Comment by Vyom-Yadav** [comment]
@Abhishek-kumar09 Why Travis CI build could not complete?

**Comment by Vyom-Yadav** [comment]
@Abhishek-kumar09  I did all the changes, please check.

**Comment by Vyom-Yadav** [comment]
github generate site

**Comment by Vyom-Yadav** [comment]
@Abhishek-kumar09 nothing is happening from the command `github generate ste`

**Comment by Abhishek-kumar09** [comment]
GitHub, generate site

**Comment by Vyom-Yadav** [comment]
> @Abhishek-kumar09 I did all the changes, please check.

@Abhishek-kumar09  Is it fine now?

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java [hunk 4]
```diff
@@ -83,6 +83,17 @@
  * <pre>
  * &lt;module name="MethodName"/&gt;
  * </pre>
+ * <p>Code Example:</p>
+ * <pre>
+ * class MyClass {
+ *   public void first_Method1() {} // OK
+ *   protected void second_Method2() {} // OK
+ *   private void Third_Method3() {} // violation, name 'ThirdMethod' must match
```
Comment (Abhishek-kumar09) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java:91]: These names doesn't align to default `MethodNameCheck`.
Since underscore is not allowed in default config.
Use should only use CamelCasing for naming methods here.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java [hunk 5]
```diff
@@ -83,6 +83,17 @@
...
+ *                                 // pattern '^[a-z](_?[a-zA-Z0-9]+)*$'
```
Comment (Abhishek-kumar09) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java:92]: This is not the default pattern.
The default pattern is:
```
"^[a-z][a-zA-Z0-9]*$"
```

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]
Items

File: src/xdocs/config_naming.xml [hunk 6]
```diff
@@ -2001,6 +2001,17 @@ class MyClass {
...
         <source>
 &lt;module name=&quot;MethodName&quot;/&gt;
+        </source>
+        <p>Code Example:</p>
+        <source>
+class MyClass {
+  public void firstMethod1() {} // OK
+  protected void secondMethod2() {} // OK
+  private void ThirdMethod3() {} // violation, name 'ThirdMethod' must match
+                                 // pattern '^[a-z][a-zA-Z0-9]*$'
```
Comment (Abhishek-kumar09) [inline src/xdocs/config_naming.xml:2011]: Please change: 
`name 'ThirdMethod' must match
                                 // pattern '^[a-z][a-zA-Z0-9]*$'` to
`method name must match to default pattern "^[a-z][a-zA-Z0-9]*$"`

**Comment by Vyom-Yadav** [comment]
@Abhishek-kumar09 why is this happening.
The Travis CI build could not complete due to an error

**Comment by Vyom-Yadav** [comment]
GitHub, generate site

File: src/xdocs/config_naming.xml [hunk 7]
```diff
@@ -2001,6 +2001,17 @@ class MyClass {
...
+  void FourthMethod4() {} // violation, name 'FourthMethod' must match
```
Comment (Abhishek-kumar09) [inline src/xdocs/config_naming.xml:2012]: This name is same as the above relative to the format. 
If you are adding one more example than add it with some different style, like you could use `_` or other similar mistakes with naming methods.
Also, add a public/protected/private modifier here.

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]
> @Abhishek-kumar09 why is this happening.
The Travis CI build could not complete due to an error

@Vyom-Yadav You could leave it for now (Some internal problems). Also add Issue number is your PR description above.

Items:

**Comment by Vyom-Yadav** [comment]
How do I generate site? still not generating.

**Comment by Abhishek-kumar09** [comment]
> How do I generate site? still not generating.

Wait a little, you have to wait for the things to be processed. Update your PR description with Issue #

**Comment by Abhishek-kumar09** [comment]
Change your 
- commit message
- PR title
- Pr description
to be more precise.

- Don't use past tense.
- `Added default config` for what? You have to write the check name also: 
 like: `add code examples for default config of MethodNameCheck`

**Comment by Vyom-Yadav** [comment]
@Abhishek-kumar09 Unable to merge this commit, what shall I do now?

**Comment by Abhishek-kumar09** [comment]
git rebase -i HEAD~4
write squash in place of pick
save and exit
git push -f

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java [hunk 8]
```diff
@@ -83,6 +83,17 @@
  * <pre>
  * &lt;module name="MethodName"/&gt;
  * </pre>
+ * <p>Code Example:</p>
+ * <pre>
+ * class MyClass {
+ *   public void firstMethod1() {} // OK
+ *   protected void secondMethod2() {} // OK
```
Comment (Abhishek-kumar09) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java:90]: remove 2 from end. It is obvious from the 1st example that you could use number with methodnaming, so rest should be more cleaner in look.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java [hunk 9]
```diff
@@ -83,6 +83,17 @@
...
+ *   private void ThirdMethod3() {} // violation, method name must match to the
```
Comment (Abhishek-kumar09) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java:91]: remove 3 from here and a space before default in the below comment.

**Review by Abhishek-kumar09 — CHANGES_REQUESTED** [review]
Items:

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java (hunk 1 above)
Comment (Abhishek-kumar09) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java]: Resolved

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java (hunk 2 above)
Comment (Abhishek-kumar09) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/ConstantNameCheck.java]: Resolved

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java (hunk 3 above)
Comment (Abhishek-kumar09) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java]: Resolved

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java (hunk 4 above)
Comment (Abhishek-kumar09) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java]: Resolved

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java (hunk 5 above)
Comment (Abhishek-kumar09) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java]: Resolved

File: src/xdocs/config_naming.xml (hunk 6 above)
Comment (Abhishek-kumar09) [reply src/xdocs/config_naming.xml]: Resolved

File: src/xdocs/config_naming.xml (hunk 7 above)
Comment (Abhishek-kumar09) [reply src/xdocs/config_naming.xml]: Resolved

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java (hunk 9 above)
Comment (Vyom-Yadav) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java]: done.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java (hunk 8 above)
Comment (Vyom-Yadav) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java]: done.

**Review by Abhishek-kumar09 — APPROVED** [review]

**Comment by romani** [comment]
Github, rebase

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java (hunk 8 above)
Comment (Abhishek-kumar09) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java]: resolved

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java (hunk 9 above)
Comment (Abhishek-kumar09) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/MethodNameCheck.java]: resolved

**Comment by romani** [comment]
Github, generate web site
